### PR TITLE
Two way relationship for Group & User

### PR DESCRIPTION
**Pull request by kalebdavis**

### CODE REVIEW

File: server/services/group.js [hunk 1]
```diff
@@ -34,8 +35,15 @@ function updateGroup(groupName, groupObject) {
...
+  let tmpGroup;
+
   return JoiHelper.validate(userId, userIdSchema)
-    .then(validatedUserObject => group.addUser(validatedUserObject));
+    .then(validatedUserObject => group.addUser(validatedUserObject))
+    .then((updatedGroup) => {
+      tmpGroup = updatedGroup;
+      return userService.addGroupByUserId(userId, updatedGroup.name);
+    })
+    .then(() => tmpGroup);
```
Comment (kalebdavis) [inline server/services/group.js]: Does this return the group?

File: server/tests/services/group.test.js [hunk 2]
```diff
@@ -132,25 +132,32 @@ describe('# Group Service', () => {
...
 
-      it('should add a user to the group by id', (done) => {
-        groupService.addUser(group, userId)
+      it('should add a two way relationship between user and group', (done) => {
+        groupService.addUser(group, user.id)
           .then((savedGroup) => {
             expect(savedGroup).to.exist;
             expect(savedGroup.name).to.equal(group.name);
-            expect(savedGroup.users).to.include(userId);
+            expect(savedGroup.users).to.include(user.id);
```
Comment (kalebdavis) [inline server/tests/services/group.test.js]: Can we add a test that the user has the group name?

File: server/services/group.js (hunk 1 above)
Comment (jarrydlee) [inline server/services/group.js]: Yes

**Comment by kalebdavis** [comment]
👍

**Pull request merged**